PowerShellTools/Classification/ClassifierServices.cs

```diff
@@ -104,22 +105,51 @@ internal IEnumerable<TagInformation<ErrorTag>> TagErrorSpans(ITextBuffer buffer,
                 yield return new TagInformation<ErrorTag>(errorSpanStart, errorSpanLength, new ErrorTag("syntax error", parseError.Message));
             }
         }
+
+        /// <summary>
+        /// Returns tag information about the errors in the buffer provided. 
+        /// </summary>
+        /// <param name="buffer">The text buffer.</param>
+        /// <param name="spanStart">The start position of the current text span.</param>
+        /// <param name="errors">The parsed errors from out-proc.</param>
+        /// <returns>Error tags consumed by VS.</returns>
+        internal IEnumerable<TagInformation<ErrorTag>> TagErrorSpans(ITextBuffer buffer, int spanStart, IEnumerable<ParseErrorItem> errors)
+        {
+            var currentSnapshot = buffer.CurrentSnapshot;
+            foreach (var parseError in errors)
+            {
+                var errorSpanStart = parseError.ExtentStartOffset + spanStart;
+                var errorSpanLength = parseError.ExtentEndOffset - parseError.ExtentStartOffset;
+                if (errorSpanStart > currentSnapshot.Length || errorSpanStart + errorSpanLength > currentSnapshot.Length) continue;
+
+                if (errorSpanLength == 0)
+                {
+                    errorSpanLength = 1;
+                    if (errorSpanStart == currentSnapshot.Length)
+                    {
+                        errorSpanStart = currentSnapshot.Length - 1;
+                    }
+                }
+
+                yield return new TagInformation<ErrorTag>(errorSpanStart, errorSpanLength, new ErrorTag("syntax error", parseError.Message));

```

Comment:
Good catch. Revised both the old and new.

From: Andre Sayre [mailto:notifications@github.com]
Sent: Thursday, January 22, 2015 2:13 PM
To: Microsoft/poshtools
Cc: Huanhuan Sun
Subject: Re: [poshtools] Support 64bit syntax by parsing errors out-of-proc (#52)


In PowerShellTools/Classification/ClassifierServices.cs<https://github.com/Microsoft/poshtools/pull/52#discussion_r23415009>:

> +            foreach (var parseError in errors)

> +            {

> +                var errorSpanStart = parseError.ExtentStartOffset + spanStart;

> +                var errorSpanLength = parseError.ExtentEndOffset - parseError.ExtentStartOffset;

> +                if (errorSpanStart > currentSnapshot.Length || errorSpanStart + errorSpanLength > currentSnapshot.Length) continue;

> +

> +                if (errorSpanLength == 0)

> +                {

> +                    errorSpanLength = 1;

> +                    if (errorSpanStart == currentSnapshot.Length)

> +                    {

> +                        errorSpanStart = currentSnapshot.Length - 1;

> +                    }

> +                }

> +

> +                yield return new TagInformation<ErrorTag>(errorSpanStart, errorSpanLength, new ErrorTag("syntax error", parseError.Message));

Are we able to use PredefinedErrorTypeNames.SyntaxError instead of the string here?
[Inside MS.VS.Text.Adornments]

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/52/files#r23415009>.